Teamb update welcome login screen

mithusingh32 opened the pull request.

Commit:
Updated splash screen, login and registration screen.

Commit:
Updated splash screen, login and registration screen.

Commit:
Updated splash screen, login and registration screen.

Added new dev profile to pom.xml

Commit:
Created a "MainScreen" with a bottom nav bar.

Commit:
Created mock-server

Removed hardcoded roles. The backend will tell the frontend which role the user has.

Created example main screen for demo. This screen is used to create features for the demo and possibly set design pattern for how the main screen should function

Commit:
Created mock-server

Removed hardcoded roles. The backend will tell the frontend which role the user has.

Created example main screen for demo. This screen is used to create features for the demo and possibly set design pattern for how the main screen should function

Commit:
Commented out fetching patient details till backend is fixed.

Home page is using hardcoded data for now.

Added More button which opens a bottom drawer which then gives users more features to select from.

Setting button now opens setting page.

Commit:
dart fix --apply

Commit:
Merge remote-tracking branch 'origin/developer' into teamb-update-welcome-login-screen

Commit:
Added class and method comments
Did clean up and some restructuring

vohrr commented on careconnect2025/frontend/mock-server/README.md:
```diff
@@ -0,0 +1,96 @@
+
+## Quick Start
+
+1. **Install dependencies:**
```
Hi, this mock-server isn't something that should be committed to the codebase. 
Its ok to use something like this for local development but it needs to be removed before this gets merged and pushed to production. 

I don't think we should be adding dependencies or other development frameworks in just to mock an api that we already have built, especially since they are not part of the approved tech stack, and are solely used for development. 

IMO if you need to mock an endpoint, it is just as simple to make a controller and endpoint that just returns ok so that  frontend development isn't blocked

mithusingh32 replied:
I'm converting this to dart. 

There are 2 use case:
1. For those who don't have the hardware to run docker, spring-boot, and frontend. 
2. Team splits work by FE/BE. The FE team doesn't need to wait on BE team to finish to start working on FE.

vohrr replied:
I think I addressed  # 2 in my original post, but I understand the use case for 1, I hadn't considered that. Converting to dart sounds like a good middle ground

Commit:
Migrated mock server from node to dart (shelf package)

mithusingh32 commented on careconnect2025/backend/core/pom.xml:
```diff
@@ -237,6 +237,39 @@
     </dependencies>
 
     <profiles>
+        <profile>
```
This will allow you to dev in intelllij. 
<img width="1378" height="1030" alt="image" src="https://github.com/user-attachments/assets/38792fbd-6800-41ac-8d1e-8d9c4d7bc58a" />

lordmitchell reviewed (approved):
After the changes were made to rely only on the assigned Technology Stack, I believe that I can approve this request. I will admit this is a lot to review and I am newer to dart so I am leaning on other reviewers to assist to make sure I don't miss anything.

MickeyMaloney-umgc reviewed (changes requested):
Do not push property file changes for backend, will conflict with other developers' branches and cause their local builds to fail

vohrr commented:
Dark mode for the welcome screen is looking a little rough, very high contrast and some of the text is barely visible and most definitely wouldn't be visible to visually impaired users. 

<img width="483" height="964" alt="image" src="https://github.com/user-attachments/assets/eb2fb757-20d8-4fff-8605-b04c16c209f6" />

vohrr commented:
The back arrow  button on Symptom tracker crashes the app to a white screen. F5 in the browser reloads to the dashboard (sort of) but not sure that would work in mobile

Commit:
#389 fix: patient dashboard correctly uses App Theme styling

Removed back button for symptom tracker screen.

mithusingh32 commented:
@vohrr -  addressed in c5d082f29b3e842ebe1e2b3b92c37b209e1607bd

Commit:
The Parent Widget error was due to having a Warp() nested inside a Container(). I changed the Wrap() to a Row()

The Parent Widget error was due to having a Warp() nested inside a Container(). I changed the Wrap() to a Row()

vohrr reviewed (approved):
There are some auth issues that I don't believe were introduced with these changes but became apparent while testing. Recommend an issue be created based on offline conversations with Sudarshan and Sukhmeet. Approving since this is primarily a UI revamp
